pkg/manage/create_docker_compose.go

```diff
@@ -81,21 +81,14 @@ func Services() (map[string]Service, error) {
 	return s, nil
 }
 
-// CreateFileType provides access to os.Create and enables mocking during testing.
-type CreateFileType func(name string) (io.WriteCloser, error)
-
-var CreateFile CreateFileType = func(name string) (io.WriteCloser, error) {
-	return os.Create(name)
-}
-
 // CreateDockerComposeYML creates a docker-compose.yml file in the current working directory
 // using a template. In remote mode it uses the GitHub API to fetch the required commit IDs
 // of all services. Else it uses relative paths to local code as provided in OpenSlides
 // main repository.
-func CreateDockerComposeYML(ctx context.Context, dataPath string, remote bool) error {
+func CreateDockerComposeYML(ctx context.Context, creator func(name string) (io.WriteCloser, error), dataPath string, remote bool) error {
```

Comment:
Suggestion1: Let creator only return an `io.Writer`. Later in this function check if the given object has a method close and only in this case use it:

```go
if closer, ok := creator.(io.Closer); ok {
  defer closer.Close()
}
```
Then you can use this function with any `Writer`. For example in the test you don't need the TestWriteCloser. You can use `bytes.Buffer` directly.

Suggestion2:
Create an interface like this:

```
type creator interface{
  create(name string) (io.Writer, error)
}
```

I think this is more idiomatic. But not necessary in your case.